src/producer.rs

```diff
@@ -0,0 +1,72 @@
+use crate::SharedFluvio;
+
+use log::debug;
+use fluvio::TopicProducer;
+use fluvio::FluvioError;
+
+use node_bindgen::derive::node_bindgen;
+use node_bindgen::core::NjError;
+use node_bindgen::core::val::JsEnv;
+use node_bindgen::core::TryIntoJs;
+use node_bindgen::sys::napi_value;
+use node_bindgen::core::JSClass;
+
+pub struct TopicProducerWrapper {
+    client: SharedFluvio,
+    topic: String,
+}
+
+impl TopicProducerWrapper {
+    pub fn new(client: SharedFluvio, topic: String) -> Self {
+        Self { client, topic }
+    }
+}
+
+impl TryIntoJs for TopicProducerWrapper {
+    fn try_to_js(self, js_env: &JsEnv) -> Result<napi_value, NjError> {
+        debug!("converting FluvioWrapper to js");
+        let new_instance = TopicProducerJS::new_instance(js_env, vec![])?;
+        debug!("instance created");
+        TopicProducerJS::unwrap_mut(js_env, new_instance)?.set_client(self.client);
+        TopicProducerJS::unwrap_mut(js_env, new_instance)?.set_topic(self.topic);
+        Ok(new_instance)
+    }
+}
+
+pub struct TopicProducerJS {
+    inner: Option<SharedFluvio>,
+    topic: Option<String>,
+}
+
+#[node_bindgen]
+impl TopicProducerJS {
+    #[node_bindgen(constructor)]
+    pub fn new() -> Self {
+        Self {
+            inner: None,
+            topic: None,
+        }
+    }
+
+    pub fn set_client(&mut self, client: SharedFluvio) {
+        self.inner.replace(client);
+    }
+
+    pub fn set_topic(&mut self, topic: String) {
+        self.topic.replace(topic);
+    }
+
+    #[node_bindgen]
+    async fn send_record(&self, data: String, partition: i32) -> Result<(), FluvioError> {
+        let topic = self.topic.clone().unwrap_or_default();
+
+        if let Some(client) = self.inner.clone() {
+            let client: TopicProducer = client.write().await.topic_producer(topic).await?;
```

Comment:
Why are we using a `RwLock` for the shared client object? Something I want to point out is that recently, the client methods switched over to all taking `&self` rather than `&mut self`, so you're doing unnecessary locking at this level by trying to lock. I think you can change `SharedClient` to be just an `Arc`.

Reply:
I think the admin client is still using `&mut self`; See example:

```rust

 pub async fn list<S, F>(&mut self, filters: F) -> Result<Vec<Metadata<S>>, FluvioError>
    where
        S: ListSpec + Encoder + Decoder,
        S::Status: Encoder + Decoder,
        F: Into<Vec<S::Filter>>,
        ListResponse: TryInto<Vec<Metadata<S>>>,
        <ListResponse as TryInto<Vec<Metadata<S>>>>::Error: Display,
    {
        use std::io::Error;
        use std::io::ErrorKind;

        let list_request = S::into_list_request(filters.into());

        let response = self.send_receive(list_request).await?;

        response
            .try_into()
            .map_err(|err| Error::new(ErrorKind::Other, format!("can't convert: {}", err)).into())
    }
```